### PR TITLE
Added automation coverage for t1 osp compute resource

### PR DESCRIPTION
**Pull request by san7ket**
Depends on Gilab PR.Don't merge untill gitlab MR 116
```
 pytest tests/foreman/cli/test_computeresource_osp.py
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.12, pytest-3.4.0, py-1.5.2, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sjagtap/PycharmProjects/robottelo, inifile:
plugins: services-1.2.1, cov-2.3.1, xdist-1.15.0, mock-1.6.3, wait-for-1.0.9
collected 9 items                                                                                                                                                                                                 
2018-02-09 18:07:06 - conftest - DEBUG - Collected 9 test cases


tests/foreman/cli/test_computeresource_osp.py ......ss.                                                                                                                                                     [100%]

=============================================================================================== 0 tests deselected ================================================================================================
====================================================================================== 7 passed, 2 skipped in 157.31 seconds ======================================================================================
```

### CODE REVIEW

File: tests/foreman/cli/test_computeresource_osp.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        :CaseImportance: Critical
+        """
+        ComputeResource.create({
+            u'name': 'cr {0}'.format(gen_string(str_type='alpha')),
+            u'provider': 'Openstack',
+            u'user': self.username,
+            u'password': self.password,
+            u'tenant': self.tenant,
+            u'url': self.current_osp_url
+        })
```
Comment (pondrejk) [inline tests/foreman/cli/test_computeresource_osp.py]: no assert?

File: tests/foreman/cli/test_computeresource_osp.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    upgrade
+)
+from robottelo.test import CLITestCase
+
+
+class OSPComputeResourceTestCase(CLITestCase):
+    """OSPComputeResource CLI tests."""
+
+    @classmethod
+    @skip_if_not_set('osp')
```
Comment (pondrejk) [inline tests/foreman/cli/test_computeresource_osp.py]: I might be wrong but doesn't this just skip the setUpClass?

File: tests/foreman/cli/test_computeresource_osp.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        })
+        self.assertTrue(comp_res['name'])
+        ComputeResource.delete({'id': comp_res['id']})
+        result = ComputeResource.exists(search=('name', comp_res['name']))
+        self.assertFalse(result)
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_osp_with_url(self):
+        """Openstack compute resource negative create with invalid values
```
Comment (ldjebran) [inline tests/foreman/cli/test_computeresource_osp.py]: Attempt to create Openstack compute resource with invalid URL

File: tests/foreman/cli/test_computeresource_osp.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+                u'user': self.username,
+                u'password': self.password,
+                u'tenant': self.tenant,
+                u'url': 'invalid url'
+            })
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_with_same_name(self):
+        """Openstack compute resource negative create with the same name
```
Comment (ldjebran) [inline tests/foreman/cli/test_computeresource_osp.py]: Attempt to create Openstack compute resource with the same name as an existing one

File: tests/foreman/cli/test_computeresource_osp.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+                u'user': self.username,
+                u'password': self.password,
+                u'tenant': self.tenant,
+                u'url': self.current_osp_url
+            })
+
+    @tier1
+    @run_only_on('sat')
+    def test_positive_update_name(self):
+        """Openstack compute resource positive update
```
Comment (ldjebran) [inline tests/foreman/cli/test_computeresource_osp.py]: Update Openstack compute resource name

File: tests/foreman/cli/test_computeresource_osp.py [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+        :id: 16eb2def-34d5-49c5-be22-88139fef7f97
+
+        :steps:
+
+            1. Create a osp compute resource
+            2. Update the name of the created compute resource
+
+        :expectedresults: Compute Resource is successfully updated
```
Comment (ldjebran) [inline tests/foreman/cli/test_computeresource_osp.py]: Compute Resource name is successfully updated

File: tests/foreman/cli/test_computeresource_osp.py [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+        the help of hostgroup.
+
+        :id: d85df090-875f-4f2b-b0f2-708efd5f50f3
+
+        :setup: Hostgroup and provisioning setup like domain, subnet etc.
+
+        :steps:
+
+            1. Create a osp compute resource.
+            2. Create a host on osp compute resource using the Hostgroup
```
Comment (ldjebran) [inline tests/foreman/cli/test_computeresource_osp.py]: Can we create the host before provisioning?
or the host is created when provisioning is successfully finished?

File: tests/foreman/cli/test_computeresource_osp.py [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+        the help of hostgroup.
+
+        :id: 644000fc-a131-4df1-89ad-897f0c741f06
+
+        :setup: Provisioning setup like domain, subnet etc.
+
+        :steps:
+
+            1. Create a osp compute resource.
+            2. Create a host on osp compute resource.
```
Comment (ldjebran) [inline tests/foreman/cli/test_computeresource_osp.py]: Can we create the host before provisioning?
or the host is created when provisioning is successfully finished?

File: tests/foreman/cli/test_computeresource_osp.py [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+        name = gen_string('utf8')
+        compute_resource = make_compute_resource({
+            u'name': name,
+            u'provider': 'Openstack',
+            u'user': self.username,
+            u'password': self.password,
+            u'tenant': self.tenant,
+            u'url': self.current_osp_url
+        })
+        self.assertEquals(compute_resource['name'], name)
```
Comment (ldjebran) [inline tests/foreman/cli/test_computeresource_osp.py]: perhaps we should add an other displayed info like "id" "provider"

File: tests/foreman/cli/test_computeresource_osp.py (hunk 1 above)
Comment (ldjebran) [reply tests/foreman/cli/test_computeresource_osp.py]: Think this should raise exception if the cmd is not successful, see the negative test

Comment (ldjebran) [reply tests/foreman/cli/test_computeresource_osp.py]: think we can add  with self.assertNotRaises(CLIReturnCodeError) to have a clean assertion on create

**Comment by san7ket** [comment]
@ldjebran Comments addressed , please re-visit

**Comment by san7ket** [comment]
```
pytest tests/foreman/cli/test_computeresource_osp.py
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.5.3, pytest-3.4.0, py-1.5.3, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sjagtap/PycharmProjects/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.22.2, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 9 items
2018-05-18 13:18:31 - conftest - DEBUG - Collected 9 test cases


tests/foreman/cli/test_computeresource_osp.py sssssssss                                                                                                                                 [100%]

===================================================================================== 0 tests deselected ======================================================================================
================================================================================= 9 skipped in 11.38 seconds ==================================================================================
```
All tests are blocked by https://bugzilla.redhat.com/show_bug.cgi?id=1579714

**Comment by ldjebran** [comment]
@rplevka please revisit as your comment seems resolved.

**Comment by rplevka** [comment]
ACK

**Pull request merged**